### PR TITLE
[Front end]: Added basic canvas with zooming functionality

### PR DESCRIPTION
**Pull request by rgreen312**

### CODE REVIEW

File: client/src/canvas/actions.ts [hunk 1]
```diff
@@ -14,6 +15,64 @@ export const fetchImageData = () => (dispatch) => {
...
+const setZoom = (f: number) => ({
+  type: ActionTypes.SetZoom,
+  payload: {
+    zoom: f
+  }
+});
+export type SetZoom = ReturnType<typeof setZoom>;
+
+export const setZoomFactor = (f: number) => (dispatch, getState) => {
+  if (f < 0) {
```
Comment (namanhkapur) [inline client/src/canvas/actions.ts]: can we add a comment or change the variable name to explain the reasoning?

File: client/src/canvas/actions.ts [hunk 2]
```diff
@@ -14,6 +15,64 @@ export const fetchImageData = () => (dispatch) => {
...
+    return;
+  }
+
+  const state = getState();
+  const ctx = getCanvasContext(state);
+  if (ctx) {
+    ctx.scale(f, f);
```
Comment (namanhkapur) [inline client/src/canvas/actions.ts]: I feel like we should call it something different than 'f'

File: client/src/canvas/components/Canvas.tsx [hunk 3]
```diff
@@ -1,23 +1,97 @@
...
+    // image.src = this.props.initialImage;
+
+
+    if (context) {
+      this.props.registerContext(context);
+    }
+
+    context!.imageSmoothingEnabled = false;
+
+    context!.fillStyle ='#000000';
```
Comment (namanhkapur) [inline client/src/canvas/components/Canvas.tsx]: reminder to change

File: client/src/canvas/components/Canvas.tsx [hunk 4]
```diff
@@ -1,23 +1,97 @@
...
           onComplete={(c) => console.log('color selected: ', c)}
-        />
+        /> */}
+        <div className='zoom-canvas' style={{ transform: `scale(${zoomFactor}, ${zoomFactor})` }}>
+          <canvas ref={this.canvasRef} />
+        </div>
+        <div className='zoom-controls'>
+          <Icon
+            type='plus-circle'
+            onClick={() => setZoomFactor(zoomFactor + 10)}
```
Comment (namanhkapur) [inline client/src/canvas/components/Canvas.tsx]: let's make the 10 another constant, use ZOOM_CHANGE_FACTOR

File: client/src/canvas/components/Canvas.tsx [hunk 5]
```diff
@@ -1,23 +1,97 @@
...
+          />
+          <Icon
+            type='minus-circle'
+            onClick={() => setZoomFactor(zoomFactor - 10)}
```
Comment (namanhkapur) [inline client/src/canvas/components/Canvas.tsx]: same comment as above

File: client/src/footer/components/Footer.scss [hunk 6]
```diff
@@ -1,8 +1,10 @@
 footer {
+  background-color: white;
   position: absolute;
   right: 0;
   bottom: 0;
   left: 0;
   padding: 1rem;
   text-align: center;
+  z-index: 10;
```
Comment (almagee16) [inline client/src/footer/components/Footer.scss]: not alphabetized

**Pull request merged**